reana_client/cli/workflow.py

```diff
@@ -272,18 +277,32 @@ def workflows_list(  # noqa: C901
 
         # Sort by given column, making sure that `None` is at the bottom of the list.
         sort_column_id = 2
-        if sort_columm_name.lower() in headers[type]:
+        if sort_columm_name.lower() == "run_number":
+            sort_column_id = headers[type].index("run_number")
+            data = sorted(
+                data,
+                key=lambda x: list(map(int, x[sort_column_id].split("."))),
+                reverse=True,
+            )
+        elif sort_columm_name.lower() in headers[type]:
             sort_column_id = headers[type].index(sort_columm_name.lower())
-        data = sorted(
-            data,
-            key=lambda x: (x[sort_column_id] is not None, x[sort_column_id]),
-            reverse=True,
-        )
+            data = sorted(
+                data,
+                key=lambda x: (x[sort_column_id] is not None, x[sort_column_id]),
+                reverse=True,
+            )
 
-        # Substitute `None` with "-"
         for row in data:
             for i, value in enumerate(row):
-                row[i] = value or "-"
+                # Substitute `None` with "-"
+                content = value or "-"
+                # Replace raw size with human-readable size if requested
+                if (
+                    headers[type][i] in workspace_size_header
+                    and human_readable_or_raw == "human_readable"
+                ):
+                    content = raw_to_human_readable[content]
```

Comment:
Might be a problem if somehow there is no value, it would try to get `raw_to_human_readable["-"]`